src/Firebase.Notifications/Android/PushNotificationReceiver.java

```diff
@@ -181,36 +183,57 @@ public void DoIt(RemoteMessage remoteMessage)
     void SpitOutNotification(RemoteMessage remoteMessage)
     {
         Context context = this;
+        int id = PushNotificationReceiver.nextID();
         Map<String,String> map = remoteMessage.getData();
         String title = map.get("title");
         String body = map.get("body");
         String sound = map.get("sound");
         String icon = map.get("icon");
+        
         String jsonStr = ToJsonString(remoteMessage);
-        Intent intent = new Intent(context, @(Activity.Package).@(Activity.Name).class);
+        Intent intent = new Intent(context, @(Activity.Package).@(Activity.Name).class); 
 
-        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
         intent.setAction(PushNotificationReceiver.ACTION);
         intent.putExtra(PushNotificationReceiver.EXTRA_KEY, jsonStr);
 
         android.app.PendingIntent pendingIntent =
-            android.app.PendingIntent.getActivity(context, 0, intent, android.app.PendingIntent.FLAG_ONE_SHOT);
+            android.app.PendingIntent.getActivity(context, 0, intent, android.app.PendingIntent.FLAG_UPDATE_CURRENT);
+       
+        NotificationManager  notificationManager = (NotificationManager)getSystemService(Context.NOTIFICATION_SERVICE);
+
+        String channel_name = "my_package_channel";
+        String channel_id = "my_package_channel_1";
+        String channel_description = "my_package_first_channel";
 
+        int importance = NotificationManager.IMPORTANCE_HIGH;
+        NotificationCompat.Builder notificationBuilder = null;
+        if(android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O)
+        {
+            NotificationChannel mChannel = notificationManager.getNotificationChannel(channel_id);
+            if (mChannel == null) {
```

Comment:
I know this project isn't completely consistent when it comes to brace-style, but it'd be nice if these two if-blocks were consistent, at least. I'm no Java expert, but whatever is idiomatic there is probably the right choice...